src/test/compile-fail/maybe-bounds-where.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+struct S1<T>(T) where (T): ?Sized;
+//~^ ERROR `?Trait` bounds are permitted only on fresh type parameters
+
+struct S2<T>(T) where u8: ?Sized;
+//~^ ERROR `?Trait` bounds are permitted only on fresh type parameters
+
+struct S3<T>(T) where &'static T: ?Sized;
+//~^ ERROR `?Trait` bounds are permitted only on fresh type parameters
+
+trait Trait<'a> {}
+
+struct S4<T>(T) where for<'a> T: ?Trait<'a>;
+//~^ ERROR `?Trait` bounds are permitted only on fresh type parameters
+
+struct S5<T>(*const T) where T: ?Trait<'static> + ?Sized;
```

Comment:
huh, apparently. I would have thought it'd be an error. Oh well, obviously pre-existing.